bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/GuideGroup.scala

```diff
@@ -0,0 +1,336 @@
+
+    val ps = f.createParamSet(ParamSetName)
+
+    getName.asScalaOpt.foreach { Pio.addParam(f, ps, "name", _) }
+
+    val tag = grp match {
+      case Initial           => AutoInitialTag
+      case Active(_)         => AutoActiveTag
+      case ManualGroup(_, _) => ManualTag
+    }
+    Pio.addParam(f, ps, "tag", tag.toString)
+
+    all.foreach { gpt => ps.addParamSet(gpt.getParamSet(f)) }
+
+    ps
+  }
+
+  def isAutomatic: Boolean =
+    grp.isAutomatic
+
+  def isManual: Boolean =
+    grp.isManual
+
+  def toManualGroup: GuideGroup =
+    GuideGroup(grp.cloneTargets.toManualGroup)
+}
+
+object GuideGroup extends ((GuideGrp) => GuideGroup) {
+  /** An initial automatic `GuideGroup`. */
+  val AutomaticInitial = GuideGroup(Initial)
+
+  /** An empty manual group. */
+  val ManualEmpty           = GuideGroup(ManualGroup("Manual Group", ==>>.empty))
+
+  val Grp: GuideGroup @> GuideGrp =
+    Lens.lensu((jGrp, sGrp) => jGrp.copy(grp = sGrp), _.grp)
+
+  val Name: GuideGroup @?> String =
+    Grp.partial >=> GuideGrp.Name
+
+  val ParamSetName = "guideGroup"
+
+  private sealed trait TypeTag
+  private case object AutoInitialTag extends TypeTag
+  private case object AutoActiveTag extends TypeTag
+  private case object ManualTag extends TypeTag
+
+  private val AllTags = List(AutoInitialTag, AutoActiveTag, ManualTag)
+
+  def fromParamSet(ps: ParamSet): GuideGroup = {
+    val name    = Pio.getValue(ps, "name", "Manual Group")
+    val targets = ps.getParamSets.asScala.toList.map { GuideProbeTargets.fromParamSet }.asImList
+
+    val typeTag = for {
+      s <- Option(Pio.getValue(ps, "tag"))
+      t <- AllTags.find(_.toString == s)
+    } yield t
+
+    typeTag.fold(createManual(name, targets)) {
+      case AutoInitialTag => AutomaticInitial
+      case AutoActiveTag  => createActive(targets)
+      case ManualTag      => createManual(name, targets)
+    }
+  }
+
+  @varargs
+  def create(name: String, targets: GuideProbeTargets*): GuideGroup =
+    createManual(name, targets.toList.asImList)
+
+  def create(name: String, targets: ImList[GuideProbeTargets]): GuideGroup =
+    createManual(name, targets)
+
+  def create(name: GemOption[String], targets: ImList[GuideProbeTargets]): GuideGroup =
+    createManual(name.getOrElse("Manual Group"), targets)
+
+  def createManual(name: String, targets: ImList[GuideProbeTargets]): GuideGroup = {
+    val m = ==>>.fromList(targets.asScalaList.flatMap { gpt =>
+      val lst    = gpt.getTargets.asScalaList
+      val zipOpt = gpt.getPrimaryIndex.asScalaOpt.map(i => lst.splitAt(i)).collect {
+        case (lefts, focus :: rights) => OptsList.focused(lefts, focus, rights)
+      }
+      zipOpt.orElse(lst.toNel.map(nel => OptsList.unfocused(nel))).strengthL(gpt.getGuider)
+    })
+
+    GuideGroup(ManualGroup(name, m))
+  }
+
+  def createActive(targets: ImList[GuideProbeTargets]): GuideGroup = {
+    val m = ==>>.fromList(targets.asScalaList.flatMap { gpt =>
```

Comment:
Yes, it is part of scalaz.  If we ever move to cats we'll need a replacement for it, `OneAnd`, `NonEmptyList`, and others.  In the meantime though, I don't think we should be shy about using parts of scalaz that we need.